specification/cognitiveservices/data-plane/AzureOpenAI/authoring/preview/2023-04-01-preview/images.json

```diff
@@ -0,0 +1,348 @@
+            }
+         },
+         "required": [
+            "id",
+            "status",
+            "created",
+            "last_action"
+         ]
+      },
+      "ImageResolution": {
+         "description": "The resolution of the image (256x256, 512x512, or 1024x1024). Smaller images are faster to generate.",
+         "type": "string",
+         "enum": [
+            "256x256",
+            "512x512",
+            "1024x1024"
+         ],
+         "default": "1024x1024"
+      },
+      "GenerateImagesPostRequest": {
+         "type": "object",
+         "properties": {
+            "prompt": {
+               "type": "string",
+               "description": "The prompt to generate an image from.",
+               "minLength": 1,
+               "maxLength": 1000
+            },
+            "n": {
+               "type": "integer",
+               "description": "The number of images to generate.",
+               "minimum": 1,
+               "default": 1
+            },
+            "size": {
+               "$ref": "#/definitions/ImageResolution"
+            }
+         },
+         "required": [
+            "prompt"
+         ]
+      },
+      "GenerateImagesPostResponse": {
+         "type": "object",
+         "properties": {
+            "status": {
+               "$ref": "#/definitions/Status"
+            }
+         },
+         "required": [
+            "id"
+         ]
+      }
+   },
+   "paths": {
+      "/images/generate": {
+         "post": {
+            "summary": "Generates a batch of images from a text caption",
+            "operationId": "Images_Generate",
+            "consumes": [
+               "application/json"
+            ],
+            "produces": [
+               "application/json"
+            ],
+            "parameters": [
+               {
+                  "$ref": "#/parameters/apiVersionQueryParameter"
+               },
+               {
+                  "in": "body",
+                  "name": "body",
+                  "required": true,
+                  "schema": {
+                    "$ref": "#/definitions/GenerateImagesPostRequest"
+                  }
+                }
+            ],
+            "responses": {
+               "202": {
+                  "description": "Created",
+                  "headers": {
+                     "Operation-Location": {
+                        "type": "string",
+                        "description": "URL to poll for the operation result."
+                     }
+                  },
+                  "schema": {
+                     "$ref": "#/definitions/GenerateImagesPostResponse"
+                  }
+               },
+               "default": {
+                  "description": "An error occurred.",
+                  "schema": {
+                    "$ref": "#/definitions/ErrorResponse"
+                  }
+               }
+            }
+         }
+      },
+      "/operations/{operationId}": {
```

Comment:
>/ [](http://example.com/codeflow?start=18&length=1)

as discussed, we should separate the routes already in this version /operations/images/generations/{id}

Reply:
As the response is the same, we decided to keep one route for the moment.